### PR TITLE
UUID fix

### PR DESCRIPTION
**Pull request by benaryorg**
For some reason the previous code chokes on an IllegalArgumentException from the UUID parser.
Since parsing a UUID is trivial but a bit time consuming it'd be nice to store the UUIDs in their proper data type from the start.
Then, although it's not exactly beautiful, we can log out the parsing exception as soon as it occurs and null the field (or rather keep the implicit null).

After this code needs to be aware of the nullable return type, but that shouldn't be an issue since the only place in which the uuid is used it is passed directly to the equals method of another UUID.

---

The above is the relevant commit message.
I don't know exactly why it's happening though.
I'd imagine any error in parsing the API response would yield errors back there and then, rather than create *Donator* instances with an empty string as uuid (the stacktrace I/we got was hinting at an empty string as uuid of some player, it happened whenever opening ones inventory, and I did not debug further).

Basically this approach switches over to a "fail early" model which also ignores errors.
I'd say "silently" but it actually does print a stacktrace, or so I'd hope.
Right now there are no such exceptions to be found in the log so it may also have been a temporary issue with the API used.
If you have any questions about our setup or the code feel free to ask.

Side note: if you could bump the version number and provide an official patched version, that'd be great, thanks

### CODE REVIEW

**Comment by Jeryn99** [comment]
Hey 👋
Thanks for this, I was going to get on it myself but you have beaten me haha

As we already know, the UUID was empty causing a crash, this sort of data should never have gotten past me, I am unsure how it managed to, but it did. On the night this happened, a few people report crashing, this was after I had made changes to the API, which I reverted quickly.

Before these crashes, a node js app would report the contents of a JSON file on my server, but adding new users this way became time-consuming, so I moved all the data over to a MySQL database, this was fine, but this is where the empty data came from.

I have removed this data, and the JSON method is still being used until I confirm that bad data doesn't make it in again, so there is no need for a new version release at the moment
